### PR TITLE
auto.FullyQualifiedStackName doc fix

### PR DESCRIPTION
**Pull request by szesch**
# Description

There is an obsolete note in the comment for `auto.FullyQualifiedStackName` regarding not using an FQDN with filestate backends which can cause some confusion since it is now supported.

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change
- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: sdk/go/auto/stack.go [hunk 1]
```diff
@@ -142,9 +142,6 @@ type Stack struct {
 // FullyQualifiedStackName returns a stack name formatted with the greatest possible specificity:
 // org/project/stack or user/project/stack
 // Using this format avoids ambiguity in stack identity guards creating or selecting the wrong stack.
-// Note that filestate backends (local file, S3, Azure Blob) do not support stack names in this
```
Comment (Frassle) [inline sdk/go/auto/stack.go:145]: Some users _are_ still on the legacy filestate structure without project support, plus while new filestate backends do support fully qualified names they are limited to the org being set to "organization", that should probably be called out here.

Comment (szesch) [reply sdk/go/auto/stack.go:145]: Should this comment stay as is but add the "organization" requirement, or remove it and replace it with the "organization" requirement?

Comment (Frassle) [reply sdk/go/auto/stack.go:145]: I'd probably update it to say "legacy filestate backends" for this sentence, and add another sentence probably before it about org having to be "organization" for filestate. But whatever you think is more clear is probably right here, being a user of the system.

Comment (szesch) [reply sdk/go/auto/stack.go:145]: I've updated the comment. Let me know what you think.

Comment (Frassle) [reply sdk/go/auto/stack.go:145]: Yeh this looks good. Definitely better than before so ✅

**Review by Frassle — APPROVED** [review]

**Pull request merged**